### PR TITLE
Towards CRM-20328 remove duplicate code, towards taking extension fro…

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
…m core

### CODE REVIEW

File: CRM/Contribute/Form/Contribution/Main.php [hunk 1]
```diff
@@ -727,9 +727,7 @@ public static function formRule($fields, $files, $self) {
         // For anonymous user check using dedupe rule
         // if user has Cancelled Membership
         if (!$memContactID) {
-          $dedupeParams = CRM_Dedupe_Finder::formatParams($fields, 'Individual');
-          $dedupeParams['check_permission'] = FALSE;
-          $ids = CRM_Dedupe_Finder::dupesByParams($dedupeParams, 'Individual');
+          $ids = CRM_Contact_BAO_Contact::getDuplicateContacts($fields, 'Individual', 'Unsupervised', array(), FALSE);
```
Comment (aydun) [inline CRM/Contribute/Form/Contribution/Main.php]: Could this be getFirstDuplicateContact() ?

File: CRM/PCP/Form/PCPAccount.php [hunk 2]
```diff
@@ -265,9 +265,7 @@ public function postProcess() {
       }
     }
 
-    $dedupeParams = CRM_Dedupe_Finder::formatParams($params, 'Individual');
-    $dedupeParams['check_permission'] = FALSE;
-    $ids = CRM_Dedupe_Finder::dupesByParams($dedupeParams, 'Individual', 'Unsupervised');
+    $ids = CRM_Contact_BAO_Contact::getDuplicateContacts($params, 'Individual', 'Unsupervised', array(), FALSE);
```
Comment (aydun) [inline CRM/PCP/Form/PCPAccount.php]: Could this be getFirstDuplicateContact() ?

**Review by aydun — COMMENTED** [review]
In CRM/Contribute/Form/Contribution/Main.php
and CRM/PCP/Form/PCPAccount.php 
could you use getFirstDuplicateContact instead?

CRM/Utils/DeprecatedUtils.php 
There is another dedupe section around line 1461 - should that be converted as well?

File: CRM/Contribute/Form/Contribution/Main.php (hunk 1 above)
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution/Main.php]: Yes, fixing

File: CRM/PCP/Form/PCPAccount.php (hunk 2 above)
Comment (eileenmcnaughton) [inline CRM/PCP/Form/PCPAccount.php]: yep

Comment (eileenmcnaughton) [inline CRM/PCP/Form/PCPAccount.php]: yes!

**Comment by eileenmcnaughton** [comment]
Thanks @aydun I've pushed up some fixes. Good spotting on the firstDuplicateContact - I didn't add that function until half way through, which I think shows!

**Comment by aydun** [comment]
Good to merge based on code review

**Pull request merged**